core/arch/arm/sm/sm_a32.S

```diff
@@ -139,9 +139,23 @@ UNWIND(	.cantunwind)
 	orr	r0, r0, #(SCR_NS | SCR_FIQ) /* Set NS and FIQ bit in SCR */
 	write_scr r0
 
+	dsb
+	isb
+
 	b	.smc_exit
 
 .smc_ret_to_sec:
+	/*
```

Comment:
Now that I look into it, I think I will drop this patch for now, I'll push it sometime after #1184 gets merged.